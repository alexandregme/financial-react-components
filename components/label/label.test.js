import React from 'react';
import { shallow, mount } from 'enzyme';
import Label from 'components/label';

describe('Label', () => {
  let mountedLabel;
  let label;

  beforeEach(() => {
    mountedLabel = shallow(<Label />);
    label = mountedLabel.find('label');
  });

  it('renders without crashing', () => {
    shallow(<Label />);
    expect(mountedLabel).toMatchSnapshot();
  });

  it('Contains a label element with custon props', () => {
    expect(label).toHaveLength(1);
    expect(label.prop('className')).toEqual('label');
    expect(label.props().htmlFor).toBeNull();
    expect(label.text()).toEqual('');
  });
});

describe('Label custom props', () => {
  let mountedLabel;
  let label;

  const props = {
    htmlFor: 'customId',
    text: 'custom label',
    children: <div />,
    className: 'custom',
  };

  beforeEach(() => {
    mountedLabel = mount(<Label {...props} />);
    label = mountedLabel.find('label');
  });

  it('should receive a prop htmlFor and show it as htmlFor', () => {
    expect(label.props().htmlFor).toEqual(props.htmlFor);
  });

  it('should receive a prop text and show it as label', () => {
    expect(label.text()).toEqual(props.text);
  });

  it('expect to add a custom className to label', () => {
    expect(label.prop('className')).toEqual('label custom');
  });

  it('expect label have a input children with custom Id', () => {
    expect(mountedLabel.props().children).toEqual(props.children);
  });
});
