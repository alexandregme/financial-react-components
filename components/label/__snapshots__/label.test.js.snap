// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label renders without crashing 1`] = `
<label
  className="label"
  htmlFor={null}
/>
`;
